homeassistant/components/smartthings/__init__.py

```diff
@@ -102,7 +103,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: SmartThingsConfigEntry)
 
     try:
         await session.async_ensure_token_valid()
-    except ClientError as err:
+    except ClientResponseError as err:
+        if err.status == HTTPStatus.BAD_REQUEST:
+            raise ConfigEntryAuthFailed("Token not valid, trigger renewal") from err
```

Comment:
would be nice to add translations for the `ConfigEntryAuthFailed` (also for the other occurrence) in a future PR